### PR TITLE
replace platform specific CI environment variables with generic ones

### LINKED ISSUE #59478: replace platform specific CI environment variables with generic ones
Per: https://github.com/pytorch/pytorch/pull/58341#discussion_r645763544
We have many CI environment variables used that are specific to platforms. these could be pre-set environment variables provided by the venders. 

This posts challenge when migrating between circle to gha. here are a list of env vars we should make platform independent changes. namely all `CIRCLECI_*` [built-in](https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables) environment variable

changing env var usage for scripts in:
  - [ ] tools/
  - [ ] .jenkins/
  - [ ] .circleci/scripts

and platform specific env-var to generic env-var conversion for CI configurations in:
  - [ ] .github/**
  - [ ] .circleci/**

cc @ezyang @seemethere @malfet @walterddr @pytorch/pytorch-dev-infra

### PR DESCRIPTION
**Pull request by atalman**
Fixes #59478

### CODE REVIEW

**Review by seemethere — DISMISSED** [review]
LGTM!

**Review by driazati — CHANGES_REQUESTED** [review]
Thanks for the fix! See the comment below for a couple extra changes we need before landing

File: tools/stats/print_test_stats.py [hunk 1]
```diff
@@ -693,13 +693,13 @@ def parse_reports(folder: str) -> Dict[str, TestFile]:
 
 def build_info() -> ReportMetaMeta:
     return {
-        "build_pr": os.environ.get("CIRCLE_PR_NUMBER", ""),
-        "build_tag": os.environ.get("CIRCLE_TAG", ""),
-        "build_sha1": os.environ.get("CIRCLE_SHA1", ""),
-        "build_base_commit": get_base_commit(os.environ.get("CIRCLE_SHA1", "HEAD")),
-        "build_branch": os.environ.get("CIRCLE_BRANCH", ""),
+        "build_pr": os.environ.get("PR_NUMBER", ""),
```
Comment (driazati) [inline tools/stats/print_test_stats.py:696]: There are a few more places that need adjusting, you can find them with a search over the codebase:

```bash
# look up all the relevant envs, ignore usage in .circleci files
git grep -E '(CIRCLE_BRANCH)|(CIRCLE_PR_NUMBER)|(CIRCLE_SHA1)|(CIRCLE_TAG)|(CIRCLE_WORKFLOW_ID)' | awk '{print $1}' | sort | uniq | grep -v '\.circleci'
```

Out of these I see these

```
.jenkins/pytorch/test.sh:PR_NUMBER=${PR_NUMBER:-${CIRCLE_PR_NUMBER:-}}
tools/stats/print_test_stats.py:
tools/stats/scribe.py:
tools/stats/upload_binary_size_to_scuba.py:
tools/testing/test_selections.py:
```

Most of these we only use in GitHub Actions, but `test_selections.py` is used in both CircleCI and GitHub Actions so it'll need some logic to look for the GHA env variable and fall back to the CircleCI env variable if it's not found (i.e. what `test.sh` with the line `sh:PR_NUMBER=${PR_NUMBER:-${CIRCLE_PR_NUMBER:-}}`)

Comment (atalman) [reply tools/stats/print_test_stats.py:696]: Sounds good @driazati will update the diff accordingly

**Pull request closed without merging**